src/afrolabs/components/kafka.clj

```diff
@@ -1332,24 +1364,24 @@
                                                                    :timestamp (t/instant (.timestamp r))}
                                                                 (seq hdrs) (assoc :headers hdrs)))))
                                                      (.poll consumer ^long poll-timeout))
-                    ;;;;;;;;;;;;;;;;;;;;;;
-                    ;; This code naively tries to determine consumer lag.
-                    ;; The problem is dealing with the timeouts in an intelligent way and complicated by the additional
-                    ;; processing time added to every .poll loop.
-                    ;; This should also probably be a strategy and not applied to every consumer.
-                    ;;;;;;;;;;;;;;;;;;;;;
-                    ;; end-offsets (consumer-end-offsets consumer)
-                    ;; current-offsets (consumer-current-offsets consumer)
-                    ;; _ (doseq [{:keys [topic partition offset]} current-offsets
-                    ;;           :let [end-offset (:offset
-                    ;;                             (first
-                    ;;                              (filter #(and (= (:partition %) partition)
-                    ;;                                            (= (:topic %) topic))
-                    ;;                                      end-offsets)))]]
-                    ;;     (prom/set (get-gauge-consumer-partition-lag {:consumer-group-id consumer-group-id
-                    ;;                                                  :topic             topic
-                    ;;                                                  :partition         partition})
-                    ;;               (- end-offset offset)))
+
+                    ;; request the consumer lag on this consumer's topic-partition assignment
+                    ;; and record it as a guage, per topic and per partition
+                    ;; ALSO - measure how long it takes to make this measurement
+                    _ (let [start-millis (System/currentTimeMillis)]
+                        (doseq [^TopicPartition assignment (.assignment consumer)
+                                :let [assignment-lag (.currentLag consumer assignment)]
+                                :when (not (.isEmpty assignment-lag))]
```

Comment:
The ConsumerClient API now exposes the `.currentLag` method, which returns an `OptionalLong` for a specific topic-partition. Here we expose that directly as a guage.